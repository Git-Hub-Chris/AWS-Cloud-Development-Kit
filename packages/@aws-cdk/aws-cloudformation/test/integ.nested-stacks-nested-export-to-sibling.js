"use strict";
/// !cdk-integ Stack1 Stack2
Object.defineProperty(exports, "__esModule", { value: true });
const sns = require("@aws-cdk/aws-sns");
const core_1 = require("@aws-cdk/core");
const app = new core_1.App();
const stack1 = new core_1.Stack(app, 'Stack1');
const stack2 = new core_1.Stack(app, 'Stack2');
const nestedUnderStack1 = new core_1.NestedStack(stack1, 'NestedUnderStack1');
const topicInNestedUnderStack1 = new sns.Topic(nestedUnderStack1, 'TopicInNestedUnderStack1');
new sns.Topic(stack2, 'TopicInStack2', {
    // topicName is too long for displayName, so just take the second part:
    // Stack1-NestedUnderStack1NestedStackNestedUnderStack1NestedStackResourceF616305B-EM64TEGA04J9-TopicInNestedUnderStack115E329C4-HEO7NLYC1AFL
    displayName: core_1.Fn.select(1, core_1.Fn.split('-', topicInNestedUnderStack1.topicName)),
});
app.synth();
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW50ZWcubmVzdGVkLXN0YWNrcy1uZXN0ZWQtZXhwb3J0LXRvLXNpYmxpbmcuanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyJpbnRlZy5uZXN0ZWQtc3RhY2tzLW5lc3RlZC1leHBvcnQtdG8tc2libGluZy50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiO0FBQUEsNEJBQTRCOztBQUU1Qix3Q0FBd0M7QUFDeEMsd0NBQTREO0FBRTVELE1BQU0sR0FBRyxHQUFHLElBQUksVUFBRyxFQUFFLENBQUM7QUFDdEIsTUFBTSxNQUFNLEdBQUcsSUFBSSxZQUFLLENBQUMsR0FBRyxFQUFFLFFBQVEsQ0FBQyxDQUFDO0FBQ3hDLE1BQU0sTUFBTSxHQUFHLElBQUksWUFBSyxDQUFDLEdBQUcsRUFBRSxRQUFRLENBQUMsQ0FBQztBQUV4QyxNQUFNLGlCQUFpQixHQUFHLElBQUksa0JBQVcsQ0FBQyxNQUFNLEVBQUUsbUJBQW1CLENBQUMsQ0FBQztBQUN2RSxNQUFNLHdCQUF3QixHQUFHLElBQUksR0FBRyxDQUFDLEtBQUssQ0FBQyxpQkFBaUIsRUFBRSwwQkFBMEIsQ0FBQyxDQUFDO0FBRTlGLElBQUksR0FBRyxDQUFDLEtBQUssQ0FBQyxNQUFNLEVBQUUsZUFBZSxFQUFFO0lBQ3JDLHVFQUF1RTtJQUN2RSw2SUFBNkk7SUFDN0ksV0FBVyxFQUFFLFNBQUUsQ0FBQyxNQUFNLENBQUMsQ0FBQyxFQUFFLFNBQUUsQ0FBQyxLQUFLLENBQUMsR0FBRyxFQUFFLHdCQUF3QixDQUFDLFNBQVMsQ0FBQyxDQUFDO0NBQzdFLENBQUMsQ0FBQztBQUVILEdBQUcsQ0FBQyxLQUFLLEVBQUUsQ0FBQyIsInNvdXJjZXNDb250ZW50IjpbIi8vLyAhY2RrLWludGVnIFN0YWNrMSBTdGFjazJcblxuaW1wb3J0ICogYXMgc25zIGZyb20gJ0Bhd3MtY2RrL2F3cy1zbnMnO1xuaW1wb3J0IHsgQXBwLCBGbiwgTmVzdGVkU3RhY2ssIFN0YWNrIH0gZnJvbSAnQGF3cy1jZGsvY29yZSc7XG5cbmNvbnN0IGFwcCA9IG5ldyBBcHAoKTtcbmNvbnN0IHN0YWNrMSA9IG5ldyBTdGFjayhhcHAsICdTdGFjazEnKTtcbmNvbnN0IHN0YWNrMiA9IG5ldyBTdGFjayhhcHAsICdTdGFjazInKTtcblxuY29uc3QgbmVzdGVkVW5kZXJTdGFjazEgPSBuZXcgTmVzdGVkU3RhY2soc3RhY2sxLCAnTmVzdGVkVW5kZXJTdGFjazEnKTtcbmNvbnN0IHRvcGljSW5OZXN0ZWRVbmRlclN0YWNrMSA9IG5ldyBzbnMuVG9waWMobmVzdGVkVW5kZXJTdGFjazEsICdUb3BpY0luTmVzdGVkVW5kZXJTdGFjazEnKTtcblxubmV3IHNucy5Ub3BpYyhzdGFjazIsICdUb3BpY0luU3RhY2syJywge1xuICAvLyB0b3BpY05hbWUgaXMgdG9vIGxvbmcgZm9yIGRpc3BsYXlOYW1lLCBzbyBqdXN0IHRha2UgdGhlIHNlY29uZCBwYXJ0OlxuICAvLyBTdGFjazEtTmVzdGVkVW5kZXJTdGFjazFOZXN0ZWRTdGFja05lc3RlZFVuZGVyU3RhY2sxTmVzdGVkU3RhY2tSZXNvdXJjZUY2MTYzMDVCLUVNNjRURUdBMDRKOS1Ub3BpY0luTmVzdGVkVW5kZXJTdGFjazExNUUzMjlDNC1IRU83TkxZQzFBRkxcbiAgZGlzcGxheU5hbWU6IEZuLnNlbGVjdCgxLCBGbi5zcGxpdCgnLScsIHRvcGljSW5OZXN0ZWRVbmRlclN0YWNrMS50b3BpY05hbWUpKSxcbn0pO1xuXG5hcHAuc3ludGgoKTtcbiJdfQ==