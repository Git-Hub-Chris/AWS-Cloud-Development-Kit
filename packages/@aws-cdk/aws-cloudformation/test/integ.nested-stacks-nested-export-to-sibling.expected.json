{
  "Stacks": [
    {
      "Resources": {
        "NestedUnderStack1NestedStackNestedUnderStack1NestedStackResourceF616305B": {
          "Type": "AWS::CloudFormation::Stack",
          "Properties": {
            "TemplateURL": {
              "Fn::Join": [
                "",
                [
                  "https://s3.",
                  {
                    "Ref": "AWS::Region"
                  },
                  ".",
                  {
                    "Ref": "AWS::URLSuffix"
                  },
                  "/",
                  {
                    "Ref": "AssetParameters9d3c9d939e58c71e78afab5fdb9ae11f733ab8e617191320654ec55ea4f6981fS3Bucket2B6102E2"
                  },
                  "/",
                  {
                    "Fn::Select": [
                      0,
                      {
                        "Fn::Split": [
                          "||",
                          {
                            "Ref": "AssetParameters9d3c9d939e58c71e78afab5fdb9ae11f733ab8e617191320654ec55ea4f6981fS3VersionKey1392150C"
                          }
                        ]
                      }
                    ]
                  },
                  {
                    "Fn::Select": [
                      1,
                      {
                        "Fn::Split": [
                          "||",
                          {
                            "Ref": "AssetParameters9d3c9d939e58c71e78afab5fdb9ae11f733ab8e617191320654ec55ea4f6981fS3VersionKey1392150C"
                          }
                        ]
                      }
                    ]
                  }
                ]
              ]
            }
          }
        }
      },
      "Outputs": {
        "ExportsOutputFnGetAttNestedUnderStack1NestedStackNestedUnderStack1NestedStackResourceF616305BOutputsStack1NestedUnderStack1TopicInNestedUnderStack136BDF841TopicNameD753D416": {
          "Value": {
            "Fn::GetAtt": [
              "NestedUnderStack1NestedStackNestedUnderStack1NestedStackResourceF616305B",
              "Outputs.Stack1NestedUnderStack1TopicInNestedUnderStack136BDF841TopicName"
            ]
          },
          "Export": {
            "Name": "Stack1:ExportsOutputFnGetAttNestedUnderStack1NestedStackNestedUnderStack1NestedStackResourceF616305BOutputsStack1NestedUnderStack1TopicInNestedUnderStack136BDF841TopicNameD753D416"
          }
        }
      },
      "Parameters": {
        "AssetParameters9d3c9d939e58c71e78afab5fdb9ae11f733ab8e617191320654ec55ea4f6981fS3Bucket2B6102E2": {
          "Type": "String",
          "Description": "S3 bucket for asset \"9d3c9d939e58c71e78afab5fdb9ae11f733ab8e617191320654ec55ea4f6981f\""
        },
        "AssetParameters9d3c9d939e58c71e78afab5fdb9ae11f733ab8e617191320654ec55ea4f6981fS3VersionKey1392150C": {
          "Type": "String",
          "Description": "S3 key for asset version \"9d3c9d939e58c71e78afab5fdb9ae11f733ab8e617191320654ec55ea4f6981f\""
        },
        "AssetParameters9d3c9d939e58c71e78afab5fdb9ae11f733ab8e617191320654ec55ea4f6981fArtifactHashBAADE149": {
          "Type": "String",
          "Description": "Artifact hash for asset \"9d3c9d939e58c71e78afab5fdb9ae11f733ab8e617191320654ec55ea4f6981f\""
        }
      }
    },
    {
      "Resources": {
        "TopicInStack27FD9238C": {
          "Type": "AWS::SNS::Topic",
          "Properties": {
            "DisplayName": {
              "Fn::Select": [
                1,
                {
                  "Fn::Split": [
                    "-",
                    {
                      "Fn::ImportValue": "Stack1:ExportsOutputFnGetAttNestedUnderStack1NestedStackNestedUnderStack1NestedStackResourceF616305BOutputsStack1NestedUnderStack1TopicInNestedUnderStack136BDF841TopicNameD753D416"
                    }
                  ]
                }
              ]
            }
          }
        }
      }
    }
  ],
  "NestedStacks": {
    "Stack1": [
      {
        "NestedUnderStack1NestedStackNestedUnderStack1NestedStackResourceF616305B": {
          "Template": {
            "Metadata": {
              "aws:cdk:path": "Stack1NestedUnderStack15F1F8640"
            },
            "Resources": {
              "TopicInNestedUnderStack115E329C4": {
                "Type": "AWS::SNS::Topic"
              }
            },
            "Outputs": {
              "Stack1NestedUnderStack1TopicInNestedUnderStack136BDF841TopicName": {
                "Value": {
                  "Fn::GetAtt": [
                    "TopicInNestedUnderStack115E329C4",
                    "TopicName"
                  ]
                }
              }
            }
          }
        }
      }
    ],
    "Stack2": []
  }
}